src/lib/voiceSearchHelper/index.ts

```diff
@@ -1,21 +1,32 @@
-const STATUS_INITIAL = 'initial';
-const STATUS_ASKING_PERMISSION = 'askingPermission';
-const STATUS_WAITING = 'waiting';
-const STATUS_RECOGNIZING = 'recognizing';
-const STATUS_FINISHED = 'finished';
-const STATUS_ERROR = 'error';
-
 export type VoiceSearchHelperParams = {
   searchAsYouSpeak: boolean;
   onQueryChange: (query: string) => void;
   onStateChange: () => void;
 };
 
+export type Status =
+  | 'initial'
+  | 'askingPermission'
+  | 'waiting'
+  | 'recognizing'
+  | 'finished'
+  | 'error';
```

Comment:
Could be replaced by an enum, so you could continue to use constants instead of duplicating the different values:

```js
enum Status {
  INITIAL = 'initial',
  ASKING_PERMISSION = 'askingPermission',
  WAITING = 'waiting',
  RECOGNIZING = 'recognizing',
  FINISHED = 'finished',
  ERROR = 'error',
}

// ...

setState({
   status: Status.WAITING,
});
```

Reply:
The advantage of the union type is that it completely disappears at runtime, [it's not the case with an enum](https://www.typescriptlang.org/play/index.html#src=enum%20EStatus%20%7B%0D%0A%20%20INITIAL%20%3D%20'initial'%2C%0D%0A%20%20ASKING_PERMISSION%20%3D%20'askingPermission'%2C%0D%0A%20%20WAITING%20%3D%20'waiting'%2C%0D%0A%20%20RECOGNIZING%20%3D%20'recognizing'%2C%0D%0A%20%20FINISHED%20%3D%20'finished'%2C%0D%0A%20%20ERROR%20%3D%20'error'%2C%0D%0A%7D%0D%0A%0D%0Aconst%20eValue%3A%20EStatus%20%3D%20EStatus.INITIAL%3B%0D%0A%0D%0A%2F**%0D%0A%20*%20---%0D%0A%20*%2F%0D%0A%0D%0Atype%20TStatus%20%3D%0D%0A%20%20%7C%20'initial'%0D%0A%20%20%7C%20'askingPermission'%0D%0A%20%20%7C%20'waiting'%0D%0A%20%20%7C%20'recognizing'%0D%0A%20%20%7C%20'finished'%0D%0A%20%20%7C%20'error'%3B%0D%0A%0D%0Aconst%20tValue%3A%20TStatus%20%3D%20'initial'%3B%20). We do have a kind of repetition with the enum too. We duplicate the different values but the compiler helps us with this if the type changes or we do a typo an error is thrown. One advantage of the enum is to be able to enumerate all the possibles values at runtime. But we don't use the feature.

Reply:
Thanks for the thorough explanation.